test/ConfigAggregatorTest.php

```diff
@@ -94,4 +96,40 @@ public function testConfigAggregatorCanLoadConfigFromCache()
         $this->assertInternalType('array', $mergedConfig);
         $this->assertEquals($expected, $mergedConfig);
     }
+
+    public function testConfigAggregatorRisesExceptionIfProcessorClassDoesNotExist()
+    {
+        $this->expectException(InvalidConfigProcessorException::class);
+        new ConfigAggregator([], null, ['NonExistentConfigProcessor']);
+    }
+
+    public function testConfigAggregatorRisesExceptionIfProcessorIsNotCallable()
+    {
+        $this->expectException(InvalidConfigProcessorException::class);
+        new ConfigAggregator([], null, [StdClass::class]);
+    }
+
+    public function testProcessorCanBeClosure()
+    {
+        $aggregator = new ConfigAggregator([], null, [
+            function (array $config) {
+                return $config + ['processor' => 'closure'];
+            },
+        ]);
+
+        $config = $aggregator->getMergedConfig();
+        $this->assertEquals(['processor' => 'closure'], $config);
+    }
+
+    public function testConfigAggregatorCanPostProcessConfiguration()
+    {
+        $aggregator = new ConfigAggregator([
+            function () {
+                return ['foo' => 'bar'];
+            },
+        ], null, [new FooPostProcessor]);
+        $mergedConfig = $aggregator->getMergedConfig();
+
+        $this->assertEquals($mergedConfig, ['foo' => 'bar', 'post-processed' => true]);
```

Comment:
Please change order of params - the first is _expected_, and the second is _actual_.